tests/scenarios/transform-complex-txn.spec.ts

```diff
@@ -29,130 +29,130 @@ describe('Complex transaction with many nested inner transactions', () => {
+        "globalStateDelta": [
           {
             "key": "",
             "value": {
               "action": 1,
               "bytes": "AAAAAAAAAAQAAAAAAhlUHw==",
-              "uint": 0,
+              "uint": 0n,
             },
           },
           {
             "key": "AA==",
             "value": {
               "action": 1,
               "bytes": "YC4Bj8ZCXdiWg6+eYEL5yV0gvi3ucnEckrGx2BQXDDIAAAAAUuN3VwAAAAAOsZeDAQAAAABS43dXAAAAAFLkB4YAAAAAAAAAAAAAAAAAAAAA/////5S/nq4AAAAAa0BhUQAAAA91+xl0AAAAAALtZZ8AAAAAAwsGTgAAAAAAAA==",
-              "uint": 0,
+              "uint": 0n,
             },
           },
           {
             "key": "AQ==",
             "value": {
               "action": 1,
               "bytes": "h2MAAAAAAAAABQAAAAAAAAAZAAAAAAAAAB6KqC3yOXMVr2XD4nTi43RC3Rv0AGIvri+ssClC+HVNQgAAAAAAAAAAAA==",
-              "uint": 0,
+              "uint": 0n,
             },
           },
         ],
         "group": "6ZssGapPFZ+DyccRludq0YjZigi05/FSeUAOFNDGGlo=",
         "id": "QLYC4KMQW5RZRA7W5GYCJ4CUVWWSZKMK2V4X3XFQYSGYCJH6LI4Q/inner/5",
-        "inner-txns": [
+        "innerTxns": [
           {
-            "arc28Events": undefined,
-            "asset-transfer-transaction": {
-              "amount": 536012365,
-              "asset-id": 1390638935,
-              "close-amount": 0,
+            "assetTransferTransaction": {
+              "amount": 536012365n,
+              "assetId": 1390638935n,
+              "closeAmount": 0n,
               "receiver": "AACCDJTFPQR5UQJZ337NFR56CC44T776EWBGVJG5NY2QFTQWBWTALTEN4A",
             },
             "balanceChanges": [
               {
                 "address": "RS7QNBEPRRIBGI5COVRWFCRUS5NC5NX7UABZSTSFXQ6F74EP3CNLT4CNAM",
                 "amount": -536012365n,
-                "assetId": 1390638935,
+                "assetId": 1390638935n,
                 "roles": [
                   "Sender",
                 ],
               },
               {
                 "address": "AACCDJTFPQR5UQJZ337NFR56CC44T776EWBGVJG5NY2QFTQWBWTALTEN4A",
                 "amount": 536012365n,
-                "assetId": 1390638935,
+                "assetId": 1390638935n,
                 "roles": [
                   "Receiver",
                 ],
               },
             ],
-            "close-rewards": 0,
-            "closing-amount": 0,
-            "confirmed-round": 35214367,
-            "fee": 0,
-            "first-valid": 35214365,
-            "inner-txns": undefined,
-            "intra-round-offset": 142,
-            "last-valid": 35214369,
-            "receiver-rewards": 0,
-            "round-time": 1705252440,
+            "closeRewards": 0n,
+            "closingAmount": 0n,
+            "confirmedRound": 35214367n,
+            "fee": 0n,
+            "firstValid": 35214365n,
+            "id": "QLYC4KMQW5RZRA7W5GYCJ4CUVWWSZKMK2V4X3XFQYSGYCJH6LI4Q/inner/6",
+            "innerTxns": [],
+            "intraRoundOffset": 148,
```

Comment:
this value has changed from 142 to 148 because I fix the mapping inner txn logic. I believe we want to calculate the `intraRoundOffset` ourselves, the existing code: https://github.com/algorandfoundation/algokit-subscriber-ts/blob/main/src/subscriptions.ts#L694. I think there was a bug with the existing implementation and the `intraRoundOffset` value isn't calculated for innerTxns.

Reply:
If that's the case we should make sure it's in the release notes as a fix.

We just need to be careful that the various indexes are correct it hurt my brain when I originally did it (so wouldn't be surprised if it wasn't completely right). I was trying to match it against what indexer API was returning so use that to check it matches correctly.